woodwork/utils.py

```diff
@@ -298,3 +298,112 @@ def _parse_logical_type(logical_type, name):
         return logical_type
     else:
         raise TypeError(f"Invalid logical type specified for '{name}'")
+
+
+def concat_columns(objs, validate_schema=True):
+    """
+    Concatenate Woodwork objects along the columns axis. There can only be one index and time index
+    set across the objects passed in. As Woodwork does not allow duplicate column names,
+    will not allow duplicate columns at concatenation.
+
+    Args:
+        objs (list[Series, DataFrame]): The Woodwork objects to be concatenated. If Woodwork
+            is not initialized on any of the objects, type inference will be performed.
+        validate_schema (bool, optional): Whether validation should be performed on the typing information
+            for the concatenated DataFrame. Defaults to True.
+
+    Returns:
+        DataFrame: A Woodwork dataframe whose typing information is also a concatenation of the input dataframes.
+    """
+    if not objs:
+        raise ValueError('No objects to concatenate')
+
+    table_name = ''
+    logical_types = {}
+    semantic_tags = {}
+    col_descriptions = {}
+    col_metadata = {}
+    table_metadata = {}
+    use_standard_tags = {}
+
+    index = None
+    time_index = None
+
+    concat_objs = []
+    # Record the typing information for all the columns that have Woodwork schemas
+    col_names_seen = set()
+    for obj in objs:
+        ww_columns = {}
+        if isinstance(obj.ww.schema, ww.table_schema.TableSchema):
+            # Raise error if there's overlap between table metadata
+            overlapping_keys = obj.ww.metadata.keys() & table_metadata.keys()
+            if overlapping_keys:
+                raise ValueError(f'Cannot resolve overlapping keys in table metadata: {overlapping_keys}')
+
+            table_metadata = {**obj.ww.metadata, **table_metadata}
+
+            # Combine table names
+            if obj.ww.name is not None:
+                if table_name:
+                    table_name += '_'
+                table_name += str(obj.ww.name)
+
+            # Cannot have multiple tables with indexes or time indexes set
+            for idx_type, obj_idx, table_idx in [('index', obj.ww.index, index),
+                                                 ('time_index', obj.ww.time_index, time_index)]:
+                if obj_idx is not None:
+                    if table_idx is None:
+                        if idx_type == 'index':
+                            index = obj_idx
+                        elif idx_type == 'time_index':
+                            time_index = obj_idx
+                    else:
+                        raise IndexError(f'Cannot have {idx_type} columns set on more than one table in objs. '
+                                         f'Please remove the {idx_type} columns from all but one table.')
```

Comment:
While I don't see any problems with this code block I find it pretty hard to follow with all the nested if conditions inside the for loop. Is there a way we could refactor this to make it more readable for the future? Maybe check index and time index settings separately instead of using the same set of code?